src/jaxsim/rbda/contacts/soft.py

```diff
@@ -126,21 +165,25 @@ def build_default_from_jaxsim_model(
         critical_damping = 2 * jnp.sqrt(K * m)
         D = ξ * critical_damping
 
-        return SoftContactsParams.build(K=K, D=D, mu=μc)
+        return SoftContactsParams.build(K=K, D=D, mu=μc, p=p, q=q)
 
-    def valid(self) -> bool:
+    def valid(self) -> jtp.BoolLike:
         """
         Check if the parameters are valid.
 
         Returns:
             `True` if the parameters are valid, `False` otherwise.
         """
 
-        return (
-            jnp.all(self.K >= 0.0)
-            and jnp.all(self.D >= 0.0)
-            and jnp.all(self.mu >= 0.0)
-        )
+        return jnp.hstack(
+            [
+                self.K >= 0.0,
+                self.D >= 0,
+                self.mu >= 0.0,
+                self.p >= 0.0,
+                self.q >= 0.0,
+            ]
+        ).all()
```

Comment:
Can we just use the built-in `all`?
```suggestion
        return all(
            [
                self.K >= 0.0,
                self.D >= 0,
                self.mu >= 0.0,
                self.p >= 0.0,
                self.q >= 0.0,
            ]
        )
```

Reply:
If we use all, I'm not sure we can use the resulting bool to raise jit-compatible exceptions. We should double check this sooner or later and align all the `valid` methods scattered around.